File: go/minimum-number-of-vertices-to-reach-all-nodes.go
func findSmallestSetOfVertices(n int, edges [][]int) []int {
    visitedVertices := make(map[int]bool)
    for i := 0; i < len(edges); i++ {
        visitedVertices[edges[i][1]] = true;
    }
    smallestSetOfVertices := []int{}
    for i := 0; i < n; i++ {
        visited, ok := visitedVertices[i]
        if ok == false || visited != true {
            smallestSetOfVertices = append(smallestSetOfVertices, i)
        }
    }
    return smallestSetOfVertices
}

File: python/minimum-number-of-vertices-to-reach-all-nodes.py
class Solution:
    def findSmallestSetOfVertices(self, n: int, edges: List[List[int]]) -> List[int]:
        visited = {edge[1] for edge in edges}
        smallest_set_of_vertices = [i for i in range(n) if i not in visited]
        return smallest_set_of_vertices
